lib/common/mock.c

```diff
@@ -179,6 +180,34 @@ __wrap_endgrent(void) {
 }
 
 
+/* fopen()
+ *
+ * If pcmk__mock_fopen is set to true, later calls to fopen() must be
+ * preceeded by:
+ *
+ *     will_return(__wrap_fopen, errno_to_set);
+ *     will_return(__wrap_fopen, FILE_ptr);
+ */
+
+bool pcmk__mock_fopen = false;
+
+FILE *
+__wrap_fopen(const char *pathname, const char *mode)
+{
+    if (pcmk__mock_fopen) {
+        FILE *retval = NULL;
+
+        errno = mock_type(int);
+        retval = mock_ptr_type(FILE *);
```

Comment:
Not a big deal, but this would be marginally more useful if it took only errno, and returned NULL for errno != 0 and `__real_fopen()` for errno == 0

Reply:
Fixed.